data/the emerald sword.txt

```diff
@@ -0,0 +1,803 @@
+			`	"Unless you have a jump drive." Rakei adds. "And you do have one, don't you?"`
+			`	"I do." You answer.`
+			`	"This is perfect! Do you have a spare jump drive? If we're going there, we're going to need one to install on the Emerald Sword and bring it back home."`
+			`	"What do I get in return?" you ask.`
+			`	"The ship, of course." Albert responds. "You've helped me so much, and you went to hell and back to get this black box decoded. We're going to spend a day or so studying the ship after we get it, making a blueprint and all that, but afterwards, it's all yours to keep! Anyway, I just can't wait to see what that ship even looks like. Give us some time to assemble a crew to help refurbish the ship, and in the meantime, get us a jump drive. Come back to us once you have it.`
+				accept
+	
+	to complete
+		never
+
+mission "The Emerald Sword 3"
+	name "The Emerald Sword"
+	description "Capture the Emerald Sword."
+	landing
+	source "Zug"
+	destination "Echoes of Tragedy"
+	cargo "Outfitting Equipment" 50
+	passengers 20
+	
+	to offer
+		has "The Emerald Sword 2: active"
+	
+	blocked `You arrive to make the preparations to capture the Emerald Sword, but the Archeologists stop you. "Captain, we're going to need 50 tons of cargo to take our materials and the jump drive, and 20 additional crew to help, including us. I'd also advise that you save enough bunks to hire some additional crew. The Emerald Sword will need some people to man it."`
+	
+	on offer
+		require "Jump Drive"
+		conversation
+			`You have a jump drive in your possession. Would you like to give it to Albert's crew?`
+			choice
+				`    (No.)`
+					defer
+				`    (Yes.)`
+			`   You give Albert a call, and arrange to meet with him to plan the next step. He shows up at your ship, greets you, and starts loading in some equipment along with the jump drive you got. He has assembled a crew of around 20 people, including him and the three Hai archeologists, now comfortable to let go of their disguise once they're back in your ship.`
+			`	"<first>, meet the people who will help us recover the Emerald Sword. I've just briefed them on the situation and what we plan to do. We're going to take some specialized tools with us, along with the Jump Drive, that will allow us to repair and outfit the ship in any world we can land on. I suggest you also hire some additional crew to man the ship once we capture it, since it's not going to pilot itself. If you're ready to go, Here are the coordinates to Zubenelhakrabi."`
+			`	You thank albert, and make your preparations for the voyage into the unknown.`
+				accept
+	
+	on accept
+		fail "The Emerald Sword 2"
+		outfit "Jump Drive" -1
+		event "zubenelhakrabi discovered"
```

Comment:
Ideally, the ship would be landed on the planet and given to the player through mission functions. Being a derelict means i have to work around stuff to prevent the player from finding out about it before it's time, and many other things. Once that function is in the game, i plan to rewrite these last missions to make a lot more sense, and "unhide" that system.

Reply:
the changes done through a quest seems to always apply only when you leave the planet. It's hardcoded. Either the system appears before you reach zug, or this "bug" happens. It would be nice to have an "apply" function to reload the data with new changes while still landed. this would allow some long multiple-ending planetary quests as well. I originally wanted to have a small adventure on watcher, but it's impossible without this apply thing

Reply:
I think that the system appearing seemingly out of no where for story purposes is fine. As for the system appearing far away if you look for it to quickly, that's just a quirk of the system. Nothing to be done about it.

Reply:
You could always add a little note to the player (Depart the planet to get updated map information.) - I've already seen one such note for the player's benefit in another quest so it's not without precedent.

A system appearing out of nowhere is perfectly fine. There are many more stars in that area of the galaxy than the hyperlanes lead to, and there's simply no reason to jump to any of the others unless you know something not widely advertised. It 'appearing' is actually kinda perfect in that sense.